stream.js

```diff
@@ -272,3 +306,22 @@ Stream.equals = function ( stream1, stream2 ) {
         return Stream.equals( stream1.tail(), stream2.tail() );
     }
 };
+Stream.iterate = function ( x, f ) {
+  return new Stream( x, function () {
+    return Stream.iterate( f( x ), f);
+  } );
+};
+Stream.cycle = function( array ) {
+  var promise_generator = function( array, index) {
+    if( index >= array.length ) index = 0;
+    return function() {
+      return new Stream( array[index], promise_generator(array, index + 1 ) );
+    };
+  };
+  return new Stream( array[0], promise_generator( array, 1 ) );
+};
+
+module.exports = Stream;
+
+},{}]},{},[1])(1)
+});
```

Comment:
I'm afraid that this file is generated by browserify and new line at end of file is not in my control.